tests/NuGet.Services.AzureSearch.FunctionalTests/BasicTests/V2SearchProtocolTests.cs

```diff
@@ -0,0 +1,266 @@
+        {
+            var results = await V2SearchAsync(new V2SearchBuilder { CountOnly = true });
+
+            Assert.NotNull(results);
+            Assert.True(results.TotalHits > 1);
+            Assert.Null(results.Data);
+        }
+
+        [Fact]
+        public async Task ResultsHonorPreReleaseField()
+        {
+            var searchTerm = "json";
+
+            var resultsWithPrerelease = await V2SearchAsync(new V2SearchBuilder { Query = searchTerm, Prerelease = true }); ;
+            Assert.NotNull(resultsWithPrerelease);
+            Assert.True(resultsWithPrerelease.Data.Count > 1);
+
+            var resultsWithoutPrerelease = await V2SearchAsync(new V2SearchBuilder { Query = searchTerm, Prerelease = false });
+            Assert.NotNull(resultsWithoutPrerelease);
+            Assert.True(resultsWithoutPrerelease.Data.Count > 1);
+
+            // The result count should be different for included prerelease results, 
+            // else it means the search term responded with same results i.e. no results have prerelease versions
+            Assert.True(resultsWithPrerelease.TotalHits > resultsWithoutPrerelease.TotalHits, 
+                $"The search term {searchTerm} does not seem to have any prerelease versions in the search index.");
+
+            var hasPrereleaseVersions = resultsWithPrerelease
+                .Data
+                .Any(x => IsPrerelease(x.Version));
+            Assert.True(hasPrereleaseVersions);
+
+            hasPrereleaseVersions = resultsWithoutPrerelease
+                .Data
+                .Any(x => IsPrerelease(x.Version));
+            Assert.False(hasPrereleaseVersions);
+        }
+
+        [Fact]
+        public async Task ResultsHonorSemverLevel()
+        {
+            var searchTerm = "packageId:BaseTestPackage.SemVer2";
+            var results = await V2SearchAsync(new V2SearchBuilder { Query = searchTerm, IncludeSemVer2 = true });
+
+            Assert.NotNull(results);
+            Assert.True(results.Data.Count >= 1);
+            var atleastOneResultWithSemVer2 = results
+                .Data
+                .Any(x => IsSemVer2(x.Version));
+
+            Assert.True(atleastOneResultWithSemVer2);
+        }
+
+        [Theory]
+        [InlineData("lastEdited", "LastEdited")]
+        [InlineData("published", "Published")]
+        [InlineData("title-asc", "Title", true)]
+        [InlineData("title-desc", "Title")]
+        public async Task ResultsAreOrderedBySpecifiedParameter(string orderBy, string property, bool reverse = false)
+        {
+            var results = await V2SearchAsync(new V2SearchBuilder { Query = "json", SortBy = orderBy });
+
+            Assert.NotNull(results);
+            Assert.True(results.Data.Count > 1);
+
+            int count = results.Data.Count < 10 ? results.Data.Count : 10;
+            var topResults = results
+                .Data
+                .Select(x => x.GetType().GetProperty(property).GetValue(x, null))
```

Comment:
Reflection breaks refactoring tools. Could you instead accept a delegate that gets the results property value? Something like:

```
[Theory]
[InlineData("lastEdited", GetLastEditedPackageData)]
...
public async Task ResultsAreOrderedBySpecifiedParameter(string orderBy, Func<V2Package, object> accessor, bool reverse = false)
{
...
}

...
private object GetLastEditedPackageData(V2Package data) => data.LastEdited;
```